librandom/uniformint_randomdev.cpp

```diff
@@ -61,10 +61,17 @@ librandom::UniformIntRandomDev::set_status( const DictionaryDatum& d )
 
   // The following test is based on
   // https://www.securecoding.cert.org/confluence/display/c/INT32-C.+Ensure+that+operations+on+signed+integers+do+not+result+in+overflow
-  // The third condition tests that we can add 1 to the difference; by the time we reach
-  // that clause, we know that the first two are false, so that the difference does not overflow.
-  if ( ( new_nmin > 0 && new_nmax < std::numeric_limits< long >::min() + new_nmin )
-    || ( new_nmin < 0 && new_nmax > std::numeric_limits< long >::max() + new_nmin )
+  //
+  // However, this one does not requre the range check for the case where
+  // the first operand is smaller than the second:
+  //    new_nmin > 0 && new_nmax < min + new_nmin
+  // because by the preceding, new_nmax >= new_nmin and thus new_nmix can not be
+  //    less than any number smaller than new_nmin
+  //
+  // The second condition tests that we can add 1 to the difference; by the time we reach
+  // that clause, we know that the first is false, so that the difference does not overflow.
```

Comment:
Better than my original, but I just went out and explicitly described all the bounds and tests.